sdk/keyvault/azure_security_keyvault_secrets/src/generated/clients/secret_client.rs

```diff
@@ -0,0 +1,520 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) Rust Code Generator. DO NOT EDIT.
+
+//cspell: words deletedsecrets maxresults
+
+use crate::models::{
+    BackupSecretResult, DeletedSecretBundle, DeletedSecretListResult, SecretBundle,
+    SecretListResult, SecretRestoreParameters, SecretSetParameters, SecretUpdateParameters,
+};
+use azure_core::credentials::TokenCredential;
+use azure_core::{
+    BearerTokenCredentialPolicy, ClientMethodOptions, ClientOptions, Context, Method, Pager,
+    Pipeline, Policy, Request, RequestContent, Response, Result, Url,
+};
+use std::sync::Arc;
+use typespec_client_core::http::PagerResult;
+use typespec_client_core::json;
+
+pub struct SecretClient {
+    api_version: String,
+    endpoint: Url,
+    pipeline: Pipeline,
+}
+
+#[derive(Clone, Debug)]
+pub struct SecretClientOptions {
+    pub api_version: String,
+    pub client_options: ClientOptions,
+}
+
+impl SecretClient {
+    pub fn new(
+        endpoint: &str,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<SecretClientOptions>,
+    ) -> Result<Self> {
+        let options = options.unwrap_or_default();
+        let mut endpoint = Url::parse(endpoint)?;
+        endpoint.set_query(None);
+        let auth_policy: Arc<dyn Policy> = Arc::new(BearerTokenCredentialPolicy::new(
+            credential,
+            vec!["https://vault.azure.net/.default"],
+        ));
+        Ok(Self {
+            endpoint,
+            api_version: options.api_version,
+            pipeline: Pipeline::new(
+                option_env!("CARGO_PKG_NAME"),
+                option_env!("CARGO_PKG_VERSION"),
+                options.client_options,
+                Vec::default(),
+                vec![auth_policy],
+            ),
+        })
+    }
+
+    /// Returns the Url associated with this client.
+    pub fn endpoint(&self) -> &Url {
+        &self.endpoint
+    }
+
+    /// Backs up the specified secret.
+    ///
+    /// Requests that a backup of the specified secret be downloaded to the client. All versions of the secret will be downloaded.
+    /// This operation requires the secrets/backup permission.
+    pub async fn backup_secret(
+        &self,
+        secret_name: String,
```

Comment:
`secret_name` and, where present, `secret_version`, should all be just `name` or `version`. THis will require a rename in the TypeSpec and is done across all languages already.

The SDK architect guidance was, where context is obvious, just use "name" or "version". Where it's not e.g., several places in `CertificateClient` that deal with things that are not a certificate e.g., issuers, only then should you preface "name".

Reply:
I'll rename these in a follow up PR